src/worker.rs

```diff
@@ -42,20 +42,22 @@ impl<'a> SidekiqWorker<'a> {
                pool: RedisPool,
                tx: Sender<Signal>,
                rx: Receiver<Operation>,
-               queues: Vec<String>,
+               mut queues: Vec<String>,
+               queue_shuffle: bool,
                handlers: BTreeMap<String, Box<dyn JobHandler>>,
                middlewares: Vec<Box<dyn MiddleWare>>,
                namespace: String)
                -> SidekiqWorker<'a> {
 
         let mut rng = rand::thread_rng();
-        let identity: Vec<u8> = iter::repeat(()).map(|()| rng.sample(distributions::Alphanumeric)).take(9).collect();
+        let identity: Vec<u8> = iter::repeat(()).map(|()| rng.sample(distributions::Alphanumeric)).take(6).collect();
 
-        // 1. randomize the queue list (which is filled with duplicates based on priority)
-        let mut queues = queues.clone();
-        queues.shuffle(&mut rng);
-        // 2. remove duplicates. this ensures that some workers will prefer low-priority queues
-        let queues = queues.into_iter().collect::<HashSet<String>>().into_iter().collect();
+        // Randomize the queue list (which is filled with duplicates based on priority)
+        // This ensures some workers will prefer low-priority queues to avoid queue starvation
+        if queue_shuffle {
+            queues.shuffle(&mut rng);
+        }
+        queues = queues.into_iter().collect::<HashSet<_>>().into_iter().collect();
```

Comment:
Do we need to deduplicate here still, given there's now [a new deduplication above](https://github.com/pganalyze/sidekiq_server.rs/pull/12/files#diff-48d27698cc708c7efe770fd897e4885efaa0a15cae30d54936068603ba03bbd9R267)?

Reply:
That's in the `report_alive` function, which reports the status to Redis. It's separately needed there so `/sidekiq/busy` doesn't show duplicate queue names.